libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/router/RouterInterfaceDelegate.kt

```diff
@@ -0,0 +1,159 @@
+package com.mapbox.navigation.navigator.internal.router
+
+import androidx.annotation.VisibleForTesting
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.navigation.base.route.RouteRefreshCallback
+import com.mapbox.navigation.base.route.RouteRefreshError
+import com.mapbox.navigation.base.route.Router
+import com.mapbox.navigation.base.route.RouterCallback
+import com.mapbox.navigation.base.route.RouterFailure
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigation.navigator.internal.mapToDirectionsResponse
+import com.mapbox.navigation.navigator.internal.mapToNativeRouteOrigin
+import com.mapbox.navigator.RouteRefreshOptions
+import com.mapbox.navigator.RouterError
+import com.mapbox.navigator.RouterErrorType
+import com.mapbox.navigator.RouterInterface
+import java.net.URL
+
+private typealias NativeRouterCallback = com.mapbox.navigator.RouterCallback
+private typealias NativeRouterRefreshCallback = com.mapbox.navigator.RouterRefreshCallback
+private typealias NativeRouterOrigin = com.mapbox.navigator.RouterOrigin
+
+class RouterInterfaceDelegate(
+    private val router: Router
```

Comment:
Why not `NavigationRouter` ?

Reply:
it's actually delegating calls from native Router to Customer one. Not sure that `NavigationRouter` is right name